web/src/features/hfiCalculator/components/PrepLevelCell.tsx

```diff
@@ -0,0 +1,107 @@
+import { makeStyles, TableCell } from '@material-ui/core'
+import { isUndefined } from 'lodash'
+import React from 'react'
+
+export interface PrepLevelCellProps {
+  testid?: string
+  meanIntensityGroup: number | undefined
+  areaName: string
+}
+
+const prepLevelColours: { [description: string]: string } = {
+  green: '#A0CD63',
+  blue: '#4CAFEA',
+  yellow: '#FFFD54',
+  orange: '#F6C142',
+  brightRed: '#EA3223',
+  bloodRed: '#B02318'
+}
+
+const useStyles = makeStyles({
+  prepLevel1: {
+    background: prepLevelColours.green,
+    fontWeight: 'bold',
+    textAlign: 'center'
+  },
+  prepLevel2: {
+    background: prepLevelColours.blue,
+    fontWeight: 'bold',
+    textAlign: 'center'
+  },
+  prepLevel3: {
+    background: prepLevelColours.yellow,
+    fontWeight: 'bold',
+    textAlign: 'center'
+  },
+  prepLevel4: {
+    background: prepLevelColours.orange,
+    fontWeight: 'bold',
+    textAlign: 'center'
+  },
+  prepLevel5: {
+    background: prepLevelColours.brightRed,
+    fontWeight: 'bold',
+    textAlign: 'center',
+    color: 'white'
+  },
+  prepLevel6: {
+    background: prepLevelColours.bloodRed,
+    fontWeight: 'bold',
+    textAlign: 'center',
+    color: 'white'
+  },
+  default: {
+    backgroundColor: 'rgba(40, 53, 147, 0.05)'
+  }
+})
+
+export type PrepLevel = 1 | 2 | 3 | 4 | 5 | 6 | undefined
+
+export const calculatePrepLevel = (meanIntensityGroup: number | undefined): PrepLevel => {
```

Comment:
I've made a ticket to move this kind of logic to the back-end. I feel like this is all stuff that should be part of the json response coming back from the API.

Reply:
It wouldn't be another network call - I'd rather envision everything being returned at once.